miasm2/core/locationdb.py

```diff
@@ -252,7 +252,7 @@ def add_location(self, name=None, offset=None, strict=True):
             # Non-strict mode
             if name_loc_key is not None:
                 known_offset = self.get_offset_location(name_loc_key)
-                if known_offset != offset:
+                if known_offset is not None and known_offset != offset:
```

Comment:
IMHO, this is not enough: if `known_offset` is actually `None`, the returned `name_loc_key` will not have its offset set, so there is a set offset to do.

Reply:
Yes! I added the set_offset.

Reply:
I agree with you: I moved the code

Reply:
Yes! I added the set_offset.